ui/src/views/Secrets/ViewSecrets/index.jsx

```diff
@@ -36,6 +37,18 @@ export default class ViewSecrets extends Component {
     secretSearch: '',
   };
 
+  static getDerivedStateFromProps(nextProps, prevState) {
+    const searchString = qs.parse(nextProps.location.search, {
+      ignoreQueryPrefix: true,
+    }).search;
+
+    if (prevState.secretSearch !== searchString) {
+      return { secretSearch: searchString };
+    }
+
+    return null;
+  }
```

Comment:
I just created https://github.com/taskcluster/taskcluster/pull/2660/commits/bd9db1a9905a204a816379468d401394e6ef740c to move away from this kind of logic 😅 See how there's a simpler way to do this without relying on internal state.